### PR TITLE
[ BB2-1152 ] Updates from utilizing the SDK in the sample app

### PR DESCRIPTION
**Pull request by dtisza1**
**JIRA Ticket:**
[BB2-1152](https://jira.cms.gov/browse/BB2-1152)

**User Story or Bug Summary:**

This improves the packaging and README.md gained from testing with the sample app.

There is a corresponding PR in the sample-client repo related to utlizing the SDK as a wheel package. See [SAMPLE APP PR here](https://github.com/CMSgov/bluebutton-sample-client-python-react/pull/15)

### What Does This PR Do?

**The following is a summary of the files in this PR**

* Updates to MANIFEST.in

* Updates to README.md

* Move source files up a level from `src/cms_bluebutton` to `cms_bluebutton`.

* Update `setup.py` to utilize `setuptools.find_packages()`. The previous version wasn't including the sub-modules in the package.

* Update README.md for packaging instruction.

* Add get/set methods to AuthorizationToken class
  - Add `get_dict()` to `AuthorizationToken` class for SDK user to get auth token config dictionary in a format that is nice for them to store. The `expires_at` is converted to an ISO date/time string format.
  - Add `set_dict()` to `AuthorizationToken` class for SDK user to be able to configure a class instance with their stored configuration values.
  - Add related tests to `test_auth.py`.

* Fix call to `refresh_auth_token(auth_token)` method in `handle_expired()` in fhir_request.py

### What Should Reviewers Watch For?

If you're reviewing this PR, please check these things, in particular:

* Are there any unhandled and/or untested edge cases you can think of?
* Does this make any backwards-incompatible changes that might break end user clients?
* Can you find any bugs if you run the code locally and test it manually?
* Naming in general? Do the namings of variables, methods, classes, and other items make sense? Are they too verbose, or not verbose enough? Any redundancies? Are the comments descriptive enough?
* Are audit logging items properly handled? Is sensitive information either not included or hashed?

* *What/how to review:*

### CODE REVIEW

**Comment by dtisza1** [comment]
Just put this PR in draft mode in case there are more changes related to testing with the sample app.

**Review by JFU-GIT — COMMENTED** [review]
not sure on python sdk repo, do we still do lint with flake8?
got following when run flake8 at python virtual env:

./cms_bluebutton/tests/test_configs.py:20:40: W504 line break after binary operator
./cms_bluebutton/tests/test_configs.py:21:14: E128 continuation line under-indented for visual indent
./cms_bluebutton/tests/test_configs.py:33:40: W504 line break after binary operator
./cms_bluebutton/tests/test_configs.py:34:14: E128 continuation line under-indented for visual indent
./cms_bluebutton/tests/test_configs.py:42:40: W504 line break after binary operator
./cms_bluebutton/tests/test_configs.py:43:14: E128 continuation line under-indented for visual indent
./cms_bluebutton/tests/test_configs.py:95:40: W504 line break after binary operator
./cms_bluebutton/tests/test_configs.py:96:14: E128 continuation line under-indented for visual indent
./cms_bluebutton/tests/fixtures/token_response.py:1:131: E501 line too long (304 > 130 characters)
./cms_bluebutton/tests/fixtures/token_response.py:2:131: E501 line too long (331 > 130 characters)
./cms_bluebutton/tests/fixtures/token_response.py:2:332: W292 no newline at end of file

**Comment by dtisza1** [comment]
> not sure on python sdk repo, do we still do lint with flake8? got following when run flake8 at python virtual env:
> 
@JFU-GIT Good call out.  We should!  I'm thinking at the end of reviewing this PR, we should run black/flake8 against all of the files. Didn't want to do that much as part of this PR, so that it's easier to see the key changes vs. formatting stuff. Was thinking we should do the same thing for the sample-client too.

File: cms_bluebutton/auth.py [hunk 1]
```diff
@@ -54,7 +76,7 @@ def refresh_auth_token(bb, auth_token):
 def generate_authorize_url(bb, auth_data):
     params = {
         "client_id": bb.client_id,
-        "redirect_uri": bb.client_secret,
+        "redirect_uri": bb.callback_url,
```
Comment (bfausett77) [inline cms_bluebutton/auth.py:79]: Good find!

File: README.md [hunk 2]
```diff
@@ -88,26 +79,59 @@ $ coverage run -m pytest
...
+  ```
+  # From repository root directory:
+  $ rm -rf build/
+  $ python setup.py sdist
+  ```
+
+The resulting distribution files with be created in the `dist/` directory.
+
+
+### Publishing
```
Comment (bfausett77) [inline README.md:137]: READ ME cleanup looks better and seems easier to install.

**Review by bfausett77 — APPROVED** [review]
LGTM

**Review by JFU-GIT — COMMENTED** [review]
May be also address the url string concat error as reported by:
https://jira.cms.gov/browse/BB2-1220
as a drive by fix?

one liner change.
thanks

**Comment by dtisza1** [comment]
> May be also address the url string concat error as reported by: https://jira.cms.gov/browse/BB2-1220 as a drive by fix?
> 
> one liner change. thanks

@JFU-GIT Good catch. This was testing OK for me, since my version was in str type.  I just added a fix for this, so that integer values (and defaults) work.   TY!

**Review by JFU-GIT — APPROVED** [review]
Tested and verified following the readme, works as expected

**Comment by dtisza1** [comment]
This is ready for additional review and testing.
